? IGConfig.xml
? mypatch.patch
? notes.txt
? testout.txt
? wxRcVC.dsp.#1.6
? wxRcVC.dsw.#1.1
Index: wxrc.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/contrib/utils/wxrc/wxrc.cpp,v
retrieving revision 1.13
diff -b -u -2 -r1.13 wxrc.cpp
--- wxrc.cpp	9 Mar 2002 16:35:00 -0000	1.13
+++ wxrc.cpp	17 May 2002 14:38:22 -0000
@@ -83,5 +83,6 @@
     static const wxCmdLineEntryDesc cmdLineDesc[] =
     {
-        { wxCMD_LINE_SWITCH, "h", "help",  "show help message" },
+        { wxCMD_LINE_SWITCH, "h", "help",  "show help message", wxCMD_LINE_VAL_STRING,
+          wxCMD_LINE_OPTION_HELP|wxCMD_LINE_PARAM_OPTIONAL  },
         { wxCMD_LINE_SWITCH, "v", "verbose", "be verbose" },
         { wxCMD_LINE_SWITCH, "c", "cpp-code",  "output C++ source rather than .rsc file" },
@@ -89,8 +90,11 @@
         { wxCMD_LINE_OPTION, "n", "function",  "C++ function name (with -c) [InitXmlResource]" },
         { wxCMD_LINE_OPTION, "o", "output",  "output file [resource.xrs/cpp]" },
-        { wxCMD_LINE_OPTION, "l", "list-of-handlers",  "output list of neccessary handlers to this file" },
+
+#if NOT_IMPLEMENTED
+        { wxCMD_LINE_OPTION, "l", "list-of-handlers",  "output list of neccessary handlers to this file" }
+#endif      
 
         { wxCMD_LINE_PARAM,  NULL, NULL, "input file(s)",
-            wxCMD_LINE_VAL_STRING, wxCMD_LINE_PARAM_MULTIPLE },
+            wxCMD_LINE_VAL_STRING, wxCMD_LINE_PARAM_MULTIPLE|wxCMD_LINE_OPTION_MANDATORY },
 
         { wxCMD_LINE_NONE }
@@ -98,8 +102,12 @@
 
 #if wxUSE_GUI
+#if !defined(WIN32)
+	// Don't use if for Windows, as the parser will put up a dialog box
+	
     // VS: I need reasonable output to console from wxCmdLineParser
     wxLog::SetTimestamp(NULL);
     delete wxLog::SetActiveTarget(new wxLogStderr);
 #endif
+#endif
 
     wxCmdLineParser parser(cmdLineDesc, argc, argv);
@@ -112,4 +120,9 @@
 
         case 0:
+            if ( parser.GetParamCount() == 0 )
+            { // no parameters isn't good either
+                parser.Usage();
+                return 0;
+            }
             retCode = 0;
             ParseParams(parser);

 	  	 
